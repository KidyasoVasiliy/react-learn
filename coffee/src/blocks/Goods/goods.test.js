import React from 'react';
import { Goods } from './index';
import { shallow } from 'enzyme';

describe('Testing <Goods />', () => {
  it('Goods have renderer correctly', () => {
    const goods = shallow(<Goods list={[{name: 'random'}]}/>);
    expect(goods).toMatchSnapshot();
  })
});
