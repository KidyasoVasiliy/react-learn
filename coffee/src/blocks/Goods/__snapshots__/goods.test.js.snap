// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <Goods /> Goods have renderer correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Goods
    list={
      Array [
        Object {
          "name": "random",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="goods__item"
        >
          <CardItem
            country={null}
            handleClick={[Function]}
            title="random"
          />
        </div>,
      ],
      "className": "goods",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "undefined-undefined",
        "nodeType": "host",
        "props": Object {
          "children": <CardItem
            country={null}
            handleClick={[Function]}
            title="random"
          />,
          "className": "goods__item",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "country": null,
            "handleClick": [Function],
            "price": undefined,
            "source": undefined,
            "title": "random",
            "transparent": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="goods__item"
          >
            <CardItem
              country={null}
              handleClick={[Function]}
              title="random"
            />
          </div>,
        ],
        "className": "goods",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "undefined-undefined",
          "nodeType": "host",
          "props": Object {
            "children": <CardItem
              country={null}
              handleClick={[Function]}
              title="random"
            />,
            "className": "goods__item",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "country": null,
              "handleClick": [Function],
              "price": undefined,
              "source": undefined,
              "title": "random",
              "transparent": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
